subliminal/core.py

```diff
@@ -703,3 +713,12 @@ def save_subtitles(video, subtitles, single=False, directory=None, encoding=None
             break
 
     return saved_subtitles
+
+
+def raise_for_status(r):
+    if r.status_code == 304:
+        raise TooManyRequests()
+    elif r.status_code == 503:
+        raise ServiceUnavailable
+    else:
+        r.raise_for_status()
```

Comment:
I agree this is not ideal to have unavailability errors go straight up as unhandled exceptions.

`r.status_code == 503` already raises HTTPError with response attribute on which you can check the status code
```python
try:
    r.raise_for_status()
except Exception as e:
    # e.reponse.status_code == 503
```
`TooManyRequests` is quite provider specific, I'm not sure this is relevant to put this in a global function.
Sometimes `raise_for_status` isn't used because one expects a redirect or something like that.

I suggest a less invasive approach: catch HTTPError with 503 status code pretty much as ServiceUnavailable and leave the TooManyRequests handling on a provider basis.
What do you think?